internal/distro/distro_test.go

```diff
@@ -353,7 +383,7 @@ func TestPipelineRepositories(t *testing.T) {
 				"*": {newStringSet([]string{"global-21", "global-22"})},
 				"os": {
 					// chain with payload repo only in the second set for the blueprint package depsolve
-					newStringSet([]string{"os-1", "os-2"}),
+					newStringSet([]string{"os-1", "os-2", "payload"}),
```

Comment:
This is a problem, the payload repo can't appear in the list for the base package set depsolves, because then we would use the user-provided repository when depsolving the base image pkg set.

Reply:
ugh you're right, with this approach I should merge them in InstantiateManifest, not initializeManifest.